File: InterviewBit/Strings/Minimum Characters required to make a String Palindromic.rb
class Solution
  # @param a : string
  # @return an integer
  def solve(a)
    return 0 if a.size < 2
    r = a.reverse
    (0..r.size-1).each { |i|
        if a.start_with?(r[i..-1])
            return 0 if i == 0
            return i
        end
    }
  end
end

File: InterviewBit/Strings/Minimum Characters required to make a String Palindromic.cpp
int Solution::solve(string A) {
    string B = A;
    reverse(B.begin(), B.end());
    
    if(A == B) { return 0; }
    
    string temp;
    for(int i=A.length()-1; i>=0; i--){
        temp += A[i];
        string x = temp + A;
        string y = x;
        reverse(y.begin(), y.end());
        if(x == y) { return temp.length(); }
    }
}
